src/main/io/osd.c

```diff
@@ -812,40 +832,68 @@ static void osdShowStats(void)
     displayClearScreen(osdDisplayPort);
     displayWrite(osdDisplayPort, 2, top++, "  --- STATS ---");
 
-    if (STATE(GPS_FIX)) {
+    if (osdConfig()->enabled_stats[OSD_STAT_ARMEDTIME]) {
+        displayWrite(osdDisplayPort, 2, top, "ARMED TIME       :");
+        tfp_sprintf(buff, "%02d:%02d", stats.armed_time / 60, stats.armed_time % 60);
+        displayWrite(osdDisplayPort, 22, top++, buff);
+    }
+
+    if (osdConfig()->enabled_stats[OSD_STAT_FLYTIME]) {
+        displayWrite(osdDisplayPort, 2, top, "FLY TIME         :");
+        tfp_sprintf(buff, "%02d:%02d", flyTime / 60, flyTime % 60);
+        displayWrite(osdDisplayPort, 22, top++, buff);
+    }
+
+    if (osdConfig()->enabled_stats[OSD_STAT_MAX_SPEED] && STATE(GPS_FIX)) {
         displayWrite(osdDisplayPort, 2, top, "MAX SPEED        :");
         itoa(stats.max_speed, buff, 10);
         displayWrite(osdDisplayPort, 22, top++, buff);
     }
 
-    displayWrite(osdDisplayPort, 2, top, "MIN BATTERY      :");
-    tfp_sprintf(buff, "%d.%1dV", stats.min_voltage / 10, stats.min_voltage % 10);
-    displayWrite(osdDisplayPort, 22, top++, buff);
-
-    displayWrite(osdDisplayPort, 2, top, "MIN RSSI         :");
-    itoa(stats.min_rssi, buff, 10);
-    strcat(buff, "%");
-    displayWrite(osdDisplayPort, 22, top++, buff);
+    if (osdConfig()->enabled_stats[OSD_STAT_MIN_BATTERY]) {
+        displayWrite(osdDisplayPort, 2, top, "MIN BATTERY      :");
```

Comment:
Ok. After rebasing on latest master to get the changed optimisation settings for F4:

Before:
```
 247429	  12156	  39752	 299337	  49149	./obj/main/betaflight_OMNIBUSF4.elf
 ```

After:
```
247533	  12156	  39752	 299441	  491b1	./obj/main/betaflight_OMNIBUSF4.elf
```

At least now the effect of the change in terms of size is consistent.

If we remove `osd.c` from the list of speed optimised sources (where it doesn't really belong), we get:

Before:
```
 243080	   8452	  33592	 285124	  459c4	./obj/main/betaflight_OMNIBUS.elf
 246406	  12168	  39756	 298330	  48d5a	./obj/main/betaflight_OMNIBUSF4.elf
```

After:
```
 242960	   8452	  33592	 285004	  4594c	./obj/main/betaflight_OMNIBUS.elf
 246286	  12168	  39756	 298210	  48ce2	./obj/main/betaflight_OMNIBUSF4.elf
```

This looks more like it should (and we'll need to do a good sort of the optimisation settings, there is way to much that is aggressively speed optimised).